astropy/modeling/tests/test_parameters.py

```diff
@@ -791,10 +791,12 @@ def wrapper2(a, b):
         # model is not None
         param._model_required = False
         model = mk.MagicMock()
-        with mk.patch.object(functools, "partial", autospec=True) as mkPartial:
```

Comment:
I don't grok what the original code was trying to do, being unfamiliar with so deep within modeling internals. This was added by @WilliamJamieson in #12232 so hopefully he can review? 🙏

Reply:
Would it be easier to use [`pytest.monkeypatch`](https://docs.pytest.org/en/stable/how-to/monkeypatch.html#monkeypatching-functions) instead of `unittest.mock`?

Reply:
No, it is simply not safe to patch this function at all (see linked issue at CPython).
My understanding is that MagicMock was merely a convenient way to implement comparison in very few lines.